pkg/kubelet/prober/worker.go

```diff
@@ -46,6 +45,9 @@ type worker struct {
 	// Describes the probe configuration (read-only)
 	spec *api.Probe
 
+	// The type of the worker.
+	probeType probeType
+
```

Comment:
I like having probeType stored on the worker, since I think it makes debugging a little easier. However, I could abstract away the 2 places I check the probeType:

- Create a prober.Probe method which takes the probeType and calls the correct version
- Create an initialValue function which returns the value to set during the initial delay

Alternatively, eliminating the probeType field from the worker would require adding fields for:

- resultsManager
- probeFunction
- initialValue

What are your thoughts on these approaches vs. what I have now?